src/app/apiconection/api.ts

```diff
@@ -0,0 +1,23 @@
+export const fetchData = async (): Promise<Article[]> => {
+    try {
+        const response = await fetch('https://jsonplaceholder.typicode.com/comments');
```

Comment:
ok another error here, please don't harcode these calls to the API, you, in this case, use this approach. 
ok , this is not good you need to separate the request in another file, you can create another directory called Async, and there you can create an API request service.
```
type RequestParams = {
    method: string;
    headers?: HeadersInit;
    body?: BodyInit | null;
};

type FetchOptions = {
    responseType: 'json' | 'blob';
};

async function fetchService<T>(
    url: string, 
    params: RequestParams, 
    options: FetchOptions
): Promise<T> {
    try {
        const response = await fetch(url, params);

        if (!response.ok) {
            throw new Error(`HTTP error! Status: ${response.status}`);
        }

        return options.responseType === 'json' ? await response.json() : await response.blob();
    } catch (error) {
        console.error('Fetching error:', error);
        throw error;
    }
}

async function getJsonData<T>(url: string): Promise<T> {
    return fetchService<T>(url, { method: 'GET' }, { responseType: 'json' });
}

// Example usage for Blob (file)
async function getFile(url: string): Promise<Blob> {
    return fetchService<Blob>(url, { method: 'GET' }, { responseType: 'blob' });
}

export { getJsonData, getFile };
```
And how to use it.
Suppose you have an API endpoint that returns user data in JSON format. You want to fetch this data and display it. Here's how you can use the getJsonData **function:**

```
import { getJsonData } from './fetchService'; 

interface UserData {
    id: number;
    name: string;
    email: string;
    // ... other user fields
}

async function fetchUserData() {
    try {
        const url = 'https://example.com/api/users'; 
        const userData = await getJsonData<UserData[]>(url);
        console.log('User Data:', userData);
        // Process and display the user data as needed
    } catch (error) {
        console.error('Error fetching user data:', error);
    }
}

// Call this function where you need to fetch the user data
fetchUserData();
```
please take this approach and refactor your code